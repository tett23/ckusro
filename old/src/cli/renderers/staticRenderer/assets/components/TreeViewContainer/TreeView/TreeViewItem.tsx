import React from 'react';
import {
  FileBuffer,
  fileBufferName,
} from '../../../../../../../models/FileBuffer';

export type Props = {
  fileBuffer: FileBuffer;
};

export default function TreeViewItem({ fileBuffer }: Props) {
  return (
    <li>
      <div>{fileBufferName(fileBuffer)}</div>
    </li>
  );
}
