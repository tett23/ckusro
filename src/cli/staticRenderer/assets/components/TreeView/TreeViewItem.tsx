import { join } from 'path';
import React, { ReactNode } from 'react';
import {
  FileBuffer,
  fileBufferName,
  replaceExt,
} from '../../../../../models/FileBuffer';
import { OutputContext } from '../../../../../models/OutputContext';

export type Props = {
  file: FileBuffer;
  context: OutputContext;
  children?: ReactNode;
};

export default function TreeViewItem({ context, file, children }: Props) {
  return (
    <li>
      <div>
        <a href={join(context.path, replaceExt(file))}>
          {fileBufferName(file)}
        </a>
      </div>
      <Children items={children} />
    </li>
  );
}

function Children({ items }: { items: ReactNode }) {
  if (items == null) {
    return null;
  }

  return <ul>{items}</ul>;
}
