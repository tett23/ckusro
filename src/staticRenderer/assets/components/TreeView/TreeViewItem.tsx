import { join } from 'path';
import React, { ReactNode } from 'react';
import { LoaderContext } from '../../../../loader';
import { CkusroFile, replaceExt } from '../../../../models/ckusroFile';

export type Props = {
  file: CkusroFile;
  context: LoaderContext;
  children?: ReactNode;
};

export default function TreeViewItem({ context, file, children }: Props) {
  return (
    <li>
      <div>
        <a href={join(context.path, replaceExt(file))}>{file.name}</a>
      </div>
      <Children items={children} />
    </li>
  );
}

function Children({ items }: { items: ReactNode }) {
  if (items == null) {
    return null;
  }

  return <ul>{items}</ul>;
}
