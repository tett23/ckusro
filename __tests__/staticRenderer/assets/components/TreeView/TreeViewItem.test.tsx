import { shallow } from 'enzyme';
import React from 'react';
import TreeViewItem from '../../../../../src/staticRenderer/assets/components/TreeView/TreeViewItem';
import { buildFile, buildLoaderContext } from '../../../../__fixtures__';

describe(TreeViewItem, () => {
  it('renders correctly', () => {
    const context = buildLoaderContext({ path: '/test' });
    const file = buildFile({ id: 'test_1' });
    const wrapper = shallow(<TreeViewItem context={context} file={file} />);

    expect(wrapper.find('Children').type === null);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when the component have children', () => {
    const context = buildLoaderContext({ path: '/test' });
    const file = buildFile({ id: 'test_1' });
    const child = buildFile({ id: 'test_2' });
    const wrapper = shallow(
      <TreeViewItem context={context} file={file}>
        <TreeViewItem context={context} file={child} />
      </TreeViewItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
