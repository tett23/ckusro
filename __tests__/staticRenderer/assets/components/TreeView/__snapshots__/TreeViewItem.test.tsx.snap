// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeViewItem renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TreeViewItem
    context={
      Object {
        "loaderConfig": Object {
          "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
          "ignore": Array [
            /\\\\\\.git/,
            /node_modules/,
          ],
        },
        "name": "test_namespace",
        "path": "/test",
        "type": "LocalLoaderContext",
      }
    }
    file={
      Object {
        "content": "[[bar.md]]",
        "fileType": "markdown",
        "id": "test_1",
        "isLoaded": true,
        "name": "foo.md",
        "namespace": "test",
        "path": "/foo.md",
        "strongDependencies": Array [
          "test:/bar.md",
        ],
        "variables": Array [],
        "weakDependencies": Array [
          "test:/bar.md",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <a
            href="/test/foo.html"
          >
            foo.md
          </a>
        </div>,
        <Children />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="/test/foo.html"
          >
            foo.md
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo.md",
            "href": "/test/foo.html",
          },
          "ref": null,
          "rendered": "foo.md",
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <a
              href="/test/foo.html"
            >
              foo.md
            </a>
          </div>,
          <Children />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/test/foo.html"
            >
              foo.md
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "foo.md",
              "href": "/test/foo.html",
            },
            "ref": null,
            "rendered": "foo.md",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`TreeViewItem renders correctly when the component have children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TreeViewItem
    context={
      Object {
        "loaderConfig": Object {
          "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
          "ignore": Array [
            /\\\\\\.git/,
            /node_modules/,
          ],
        },
        "name": "test_namespace",
        "path": "/test",
        "type": "LocalLoaderContext",
      }
    }
    file={
      Object {
        "content": "[[bar.md]]",
        "fileType": "markdown",
        "id": "test_1",
        "isLoaded": true,
        "name": "foo.md",
        "namespace": "test",
        "path": "/foo.md",
        "strongDependencies": Array [
          "test:/bar.md",
        ],
        "variables": Array [],
        "weakDependencies": Array [
          "test:/bar.md",
        ],
      }
    }
  >
    <TreeViewItem
      context={
        Object {
          "loaderConfig": Object {
            "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
            "ignore": Array [
              /\\\\\\.git/,
              /node_modules/,
            ],
          },
          "name": "test_namespace",
          "path": "/test",
          "type": "LocalLoaderContext",
        }
      }
      file={
        Object {
          "content": "[[bar.md]]",
          "fileType": "markdown",
          "id": "test_2",
          "isLoaded": true,
          "name": "foo.md",
          "namespace": "test",
          "path": "/foo.md",
          "strongDependencies": Array [
            "test:/bar.md",
          ],
          "variables": Array [],
          "weakDependencies": Array [
            "test:/bar.md",
          ],
        }
      }
    />
  </TreeViewItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <a
            href="/test/foo.html"
          >
            foo.md
          </a>
        </div>,
        <Children
          items={
            <TreeViewItem
              context={
                Object {
                  "loaderConfig": Object {
                    "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
                    "ignore": Array [
                      /\\\\\\.git/,
                      /node_modules/,
                    ],
                  },
                  "name": "test_namespace",
                  "path": "/test",
                  "type": "LocalLoaderContext",
                }
              }
              file={
                Object {
                  "content": "[[bar.md]]",
                  "fileType": "markdown",
                  "id": "test_2",
                  "isLoaded": true,
                  "name": "foo.md",
                  "namespace": "test",
                  "path": "/foo.md",
                  "strongDependencies": Array [
                    "test:/bar.md",
                  ],
                  "variables": Array [],
                  "weakDependencies": Array [
                    "test:/bar.md",
                  ],
                }
              }
            />
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="/test/foo.html"
          >
            foo.md
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo.md",
            "href": "/test/foo.html",
          },
          "ref": null,
          "rendered": "foo.md",
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": <TreeViewItem
            context={
              Object {
                "loaderConfig": Object {
                  "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
                  "ignore": Array [
                    /\\\\\\.git/,
                    /node_modules/,
                  ],
                },
                "name": "test_namespace",
                "path": "/test",
                "type": "LocalLoaderContext",
              }
            }
            file={
              Object {
                "content": "[[bar.md]]",
                "fileType": "markdown",
                "id": "test_2",
                "isLoaded": true,
                "name": "foo.md",
                "namespace": "test",
                "path": "/foo.md",
                "strongDependencies": Array [
                  "test:/bar.md",
                ],
                "variables": Array [],
                "weakDependencies": Array [
                  "test:/bar.md",
                ],
              }
            }
          />,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <a
              href="/test/foo.html"
            >
              foo.md
            </a>
          </div>,
          <Children
            items={
              <TreeViewItem
                context={
                  Object {
                    "loaderConfig": Object {
                      "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
                      "ignore": Array [
                        /\\\\\\.git/,
                        /node_modules/,
                      ],
                    },
                    "name": "test_namespace",
                    "path": "/test",
                    "type": "LocalLoaderContext",
                  }
                }
                file={
                  Object {
                    "content": "[[bar.md]]",
                    "fileType": "markdown",
                    "id": "test_2",
                    "isLoaded": true,
                    "name": "foo.md",
                    "namespace": "test",
                    "path": "/foo.md",
                    "strongDependencies": Array [
                      "test:/bar.md",
                    ],
                    "variables": Array [],
                    "weakDependencies": Array [
                      "test:/bar.md",
                    ],
                  }
                }
              />
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/test/foo.html"
            >
              foo.md
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "foo.md",
              "href": "/test/foo.html",
            },
            "ref": null,
            "rendered": "foo.md",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": <TreeViewItem
              context={
                Object {
                  "loaderConfig": Object {
                    "enable": /\\\\\\.\\(md\\|txt\\)\\$/,
                    "ignore": Array [
                      /\\\\\\.git/,
                      /node_modules/,
                    ],
                  },
                  "name": "test_namespace",
                  "path": "/test",
                  "type": "LocalLoaderContext",
                }
              }
              file={
                Object {
                  "content": "[[bar.md]]",
                  "fileType": "markdown",
                  "id": "test_2",
                  "isLoaded": true,
                  "name": "foo.md",
                  "namespace": "test",
                  "path": "/foo.md",
                  "strongDependencies": Array [
                    "test:/bar.md",
                  ],
                  "variables": Array [],
                  "weakDependencies": Array [
                    "test:/bar.md",
                  ],
                }
              }
            />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
